docs/modules/ROOT/pages/guides/v3-migration/index.adoc

```diff
@@ -0,0 +1,159 @@
+[[v3-migration]]
+= 3.0.0 Migration
+This document lists all breaking changes from version 2.x.y to 3.0.0 and how to update.
+
+== How to upgrade
+Simply update `@neo4j/graphql` using npm or your package manager of choice:
+
+[source, bash, indent=0]
+----
+npm update @neo4j/graphql
+----
+
+== Async schema generation
+Schema generation is now asynchronous. Instead of using the property `schema`, now the method `getSchema` will return the schema
+as a _Promise_. This means that creating a server now requires awaiting for that method:
+
+Instead of
+[source, JavaScript, indent=0]
+----
+const neoSchema = new Neo4jGraphQL({ typeDefs, driver });
+const server = new ApolloServer({
+    schema: neoSchema.schema,
+});
+----
+
+Now you'll need to do the following:
+
+[source, JavaScript, indent=0]
+----
+const neoSchema = new Neo4jGraphQL({ typeDefs, driver });
+const schema = await neoSchema.getSchema();
+const server = new ApolloServer({
+    schema: schema
+});
+----
+
+== Relationship changes
+This release contains an overhaul of our relationship validations, this will require a few changes to the schema.
+
+=== Many to * relationships
+To improve consistency and validation, **all** "many to *" relationships need to be defined as _required_ in the schema:
+
+[source, graphql, indent=0]
+----
+type Movie {
+    actors: [Actor!]! @relationship(type: "ACTED_IN", direction: IN)
+    director: Director @relationship(type: "DIRECTED", direction: IN)
+}
+----
+
+Note that any other notation, such as `[Actor]` or `[Actor!]` will **not** be valid. "One to one" relationships
+such as `Director` remain unchanged.
+
+=== Relationship cardinality
```

Comment:
Are there any more details available of the effect this will have? e.g. will the server start to error if a rogue relationship is created in the DB? Can this functionality be turned off?

Reply:
Hi @wheresrhys, we will try to be as comprehensive with potential pitfalls as possible. As you mention, there is a change the server may return an error on **mutations**  if your database has rogue relationships. Please keep in mind that if that is the case, the current behaviour is already undetermined and **will** cause problems, at least for now we do not plan on making this optional, as the current behaviour is actually buggy.